repository/src/lib.rs

```diff
@@ -146,9 +146,157 @@ impl<T: RawRepository> DistributedRepository<T> {
+            self.raw
+                .add_remote("yoonho".into(), "github".into())
+                .await?;
+
+            // TODO: change yoonho, github into something else with peer
+            // It will be evaluated by "git" method
+        }
+
+        self.raw.fetch_all().await?;
+
+        let branches_after_fetch = self.raw.list_branches().await?;
+
+        // Make block_header_vec to contain BlockHeaders below,
+        // and make block_commit_hash_vec to contain CommitHash
+        let mut block_header_vec: Vec<BlockHeader> = vec![];
+        let mut finalized_branches: Vec<Branch> = vec![];
+
+        // For all incoming branches
+        for branch in branches_after_fetch {
+            // Delete branches which merge base is not the known finalized commit
+            let branch_commit_hash = self.raw.locate_branch(branch.clone()).await?;
+            let finalized_branch_commit_hash =
+                self.raw.locate_branch(FINALIZED_BRANCH_NAME.into()).await?;
+
+            if finalized_branch_commit_hash
+                != self
+                    .raw
+                    .find_merge_base(branch_commit_hash, finalized_branch_commit_hash)
+                    .await?
+            {
+                self.raw.delete_branch(branch.to_string()).await?;
+                // Jump to the next branch
+                continue;
+            }
+
+            // Make new CSV for the branch
+            let blockheader = self.get_last_finalized_block_header().await?;
+            let branch_reserved_state = self.get_reserved_state().await?;
+            let mut branch_csv =
+                verify::CommitSequenceVerifier::new(blockheader.clone(), branch_reserved_state)
+                    .unwrap();
+
+            // Find new commits (after the finalized commit)
+            let branch_semantic_commit = self.raw.read_semantic_commit(branch_commit_hash).await?;
+            let _branch_commit =
+                format::from_semantic_commit(branch_semantic_commit).map_err(|e| anyhow!(e))?;
+
+            let branch_ancestors = self.raw.list_ancestors(branch_commit_hash, None).await?;
+            let finalized_ancestors = self
+                .raw
+                .list_ancestors(finalized_branch_commit_hash, None)
+                .await?;
+
+            let new_commits_hashes: Vec<CommitHash> = branch_ancestors
+                .into_iter()
+                .filter(|hash| !finalized_ancestors.contains(hash))
+                .collect();
+
+            let new_commits_hashes_cloned = new_commits_hashes.clone();
+            let tip_commit_hash = new_commits_hashes_cloned.last().unwrap();
+            let tip_semantic_commit = self.raw.read_semantic_commit(*tip_commit_hash).await?;
+            let tip_commit =
+                format::from_semantic_commit(tip_semantic_commit).map_err(|e| anyhow!(e))?;
+
+            for new_commit_hash in new_commits_hashes {
+                // Verify with CSV about every new commits
```

Comment:
```suggestion
            // Verify all the incoming commits and apply them to the local repository only if they are valid.
            for new_commit_hash in new_commits_hashes {
```

1. I think comments should be above the for loop describing what happens in the loop
2. I think comments inside the function should be written with similar wordings written above the function declaration.
https://github.com/postech-dao/simperby/pull/147/files#diff-4f894bfd4ede2d1934ab3833d5e0fa29e4e42df1c1fe43f3be66e9212b1cf010R138-R139